services/user/Profiles/plugin/src/lib.rs

```diff
@@ -0,0 +1,102 @@
+#[allow(warnings)]
+mod bindings;
+
+use bindings::exports::profiles::plugin::api::Guest as Api;
+use bindings::exports::profiles::plugin::contacts::Guest as Contacts;
+
+use bindings::exports::profiles::plugin::contacts::Contact;
+
+use bindings::profiles::plugin::types::Profile as PluginProfile;
+
+use bindings::exports::profiles::plugin::api::File;
+
+use bindings::accounts::plugin::api::{get_account, get_current_user};
+use bindings::host::common::client as Client;
+use bindings::host::common::types::Error;
+use bindings::transact::plugin::intf::add_action_to_transaction;
+
+use psibase::fracpack::Pack;
+
+mod errors;
+use errors::ErrorType;
+
+mod contact_table;
+
+use contact_table::ContactTable;
+
+struct ProfilesPlugin;
+
+fn check_app_sender() -> Result<(), ErrorType> {
+    match Client::get_sender_app().app {
+        Some(app) if app == "homepage" => Ok(()),
+        _ => Err(ErrorType::UnauthorizedApp()),
+    }
+}
+
+fn check_account_exists(account: &str) -> Result<(), ErrorType> {
+    get_account(account)
+        .map_err(|_| ErrorType::InvalidAccountNumber(account.to_string()))?
```

Comment:
Don't need this `map_err`, you can just use `?`

Reply:
It appears `get_account` on an error returns a `Error` not an `ErrorType`, not sure how to tackle this.

Reply:
```
fn check_app_sender() -> Result<(), Error> {
    match Client::get_sender_app().app {
        Some(app) if app == "homepage" => Ok(()),
        _ => Err(ErrorType::UnauthorizedApp().into()),
    }
}

fn check_account_exists(account: &str) -> Result<(), Error> {
    get_account(account)?
        .ok_or(ErrorType::NoAccountFound(account.to_string()).into())
        .map(|_| ())
}

fn user() -> Result<String, Error> {
    get_current_user()?.ok_or(ErrorType::NoUserLoggedIn().into())
}
```